src/Microsoft.Azure.SignalR.Management/ServiceManagerOptions.cs

```diff
@@ -34,51 +32,90 @@ public class ServiceManagerOptions
         public IWebProxy Proxy { get; set; }
 
         /// <summary>
-        /// Gets or sets the service endpoint for accessing Azure SignalR Service.
+        /// Gets or sets a service endpoint of Azure SignalR Service.
+        /// </summary>
+        public ServiceEndpoint ServiceEndpoint { get; set; }
+
+        /// <summary>
+        /// <para>Gets service endpoint(s) of Azure SignalR Service set by any one of the properties 'ConnectionString', 'ServiceEndpoint' or 'ServiceEndpoints'.</para>
+        /// <para>Sets multiple service endpoints of Azure SignalR Service.</para>
         /// </summary>
-        public ServiceEndpoint ServiceEndpoint
+        internal ServiceEndpoint[] ServiceEndpoints
         {
             get
             {
-                if (_serviceEndpoint == null)
+                if (_serviceEndpoints != null)
+                {
+                    return _serviceEndpoints;
+                }
+                if (ServiceEndpoint != null)
+                {
+                    return new ServiceEndpoint[] { ServiceEndpoint };
```

Comment:
this creates a new instance for every `get`

Reply:
How about adding it to the Context option based on the input Options to avoid changing this Options class?